File: Prob_1768_MergeStringsAlternatively/mergealternatively.java
public class MergeStringsAlternately {
    public static String mergeAlternately(String word1, String word2) {
        StringBuilder result = new StringBuilder();
        int minLength = Math.min(word1.length(), word2.length());
        
        // Merge the characters in alternating order
        for (int i = 0; i < minLength; i++) {
            result.append(word1.charAt(i));
            result.append(word2.charAt(i));
        }
        
        // Append any remaining characters from the longer word
        if (minLength < word1.length()) {
            result.append(word1.substring(minLength));
        }
        if (minLength < word2.length()) {
            result.append(word2.substring(minLength));
        }
        
        return result.toString();
    }

    public static void main(String[] args) {
        String word1 = "abc";
        String word2 = "pqr";
        System.out.println(mergeAlternately(word1, word2));  // Output: "apbqcr"
        
        word1 = "ab";
        word2 = "pqrs";
        System.out.println(mergeAlternately(word1, word2));  // Output: "apbqrs"
    }
}

File: Prob_1768_MergeStringsAlternatively/mergeAlternatively.cpp
#include <iostream>
#include <string>
using namespace std;

string mergeAlternately(string word1, string word2) {
    string result;
    int minLength = min(word1.length(), word2.length());
    
    // Merge the characters in alternating order
    for (int i = 0; i < minLength; ++i) {
        result += word1[i];
        result += word2[i];
    }
    
    // Append any remaining characters from the longer word
    if (minLength < word1.length()) {
        result += word1.substr(minLength);
    }
    if (minLength < word2.length()) {
        result += word2.substr(minLength);
    }
    
    return result;
}

int main() {
    string word1 = "abc";
    string word2 = "pqr";
    cout << mergeAlternately(word1, word2) << endl;  // Output: "apbqcr"
    
    word1 = "ab";
    word2 = "pqrs";
    cout << mergeAlternately(word1, word2) << endl;  // Output: "apbqrs"
    
    return 0;
}
